### DISCUSSION TITLE
Fixing initial distribution for one state

### DISCUSSION THREAD

**Discussion by rociojoo** [bmcclintock/momentuHMM#90] [Q&A] [answered]
Hi Brett, I'd like to forbid the state sequence to start at a certain state, so fix it's delta at 0, while still allowing for the estimation of the initial probabilities of the other states. I tried fixPar = list(delta=c(NA,0,NA)), but I get the error of:  fixPar$delta must sum to 1.
I also tried, without fixing anything, giving initial values like delta_0 <- c(0.70,0.001,0.299). It still gives me a probability close to 1 of starting at the state it shouldn't. Any thoughts?
- Rocío

**Comment by bmcclintock** [comment]
Hi Rocio,

If you know with certainty that it cannot start in state 2 but the estimates are suggesting otherwise, then this is likely an indication that the model is missing something.  How do you know that it cannot start in state 2?  Can this information be incorporated into the model as a covariate or additional data stream?  If you can explicitly incorporate this information, then I'd suggest trying that first.  

If that isn't feasible, then you can force this behavior by using a "dummy" covariate on delta.  When including a covariate on delta using 'formulaDelta', then 'delta0' and 'fixPar$delta' are on the working scale.  Unlike fixing parameters for delta on the real scale (where all must be fixed and sum to 1), no such constraints are imposed when using fixPar on the working scale parameters.  

To do this you would add a dummy covariate to your data (e.g., > data$dummy <- rnorm(nrow(data))), set formulaDelta = ~dummy, and set fixPar$delta <- matrix(c(-100,0,NA,0),2,nbStates-1,dimnames=list(c("(Intercept)","dummy"),c("state 2","state 3"))).  A sufficiently large negative value for the state 2 intercept term will effectively make the initial state probability for state 2 equal to zero. 

Cheers,
Brett

**Comment by bmcclintock** [comment, marked as answer]
Hi again Rocio,

While the suggested "trick" above works, it's a bit clunky.  I've therefore changed the default behavior for momentuHMM >=1.4.3 (currently only available in the 'develop' branch) such that when formulaDelta=NULL (the default) then delta0 and fixPar$delta are specified on the real scale.  However, whenever a formula is specified for formulaDelta (e.g. formulaDelta=~1, formulaDelta=~cov1), then delta0 and fixPar$delta are specified on the working scale.  Thus in momentuHMM >=1.4.3, one can accomplish this without needing a dummy covariate by specifying formulaDelta=~1 and fixPar$delta <- c(-100,NA), where a sufficiently large negative value for the state 2 intercept effectively makes the initial state probability for state 2 equal to zero.

Cheers,
Brett

**Comment by rociojoo** [comment]
Hi Brett, 

Thanks! I just did what you suggested in the last comment. So, just to be sure, does the function always assume that the values given for delta are from the 2nd to last state? Also, I know that this is just a trick; I agree with you about trying to deal with not starting at state 2 in other ways, I just need to give it more thought.

Cheers,
Rocío

**Comment by bmcclintock** [comment]
If formulaDelta includes a formula, then delta0 is specified on the working scale as a k x (nbStates-1) matrix, where k is the number of covariates and the columns correspond to states 2:nbStates.  The initial distribution parameters must sum to 1 on the real scale, so that means there are nbStates-1 working scale parameters (with state 1 as the reference parameter).  For more details, see ?fitHMM.

**Comment by rociojoo** [comment]
Got it, thanks!
